### ISSUE TITLE
Spelling corrections in proofreading

### ISSUE THREAD

**Issue by vr8hub** [kingrpaul/algernon-blackwood_short-fiction#11]
I hadn't looked at your commits since my initial look until today. I see that there are a number of spelling corrections mixed in the "proofreading" commits. And as I look back, I see a "modernize spellings" commit several commits back.

Per the [Step-by-Step](https://standardebooks.org/contribute/producing-an-ebook-step-by-step) guide, `se modernize-spelling` and the individual spelling fixes (e.g. any one to anyone, every one to everyone, etc.) are to be made in commits prefixed with "[Editorial]". From that guide: "After you’ve reviewed the changes that the tool made, create an “[Editorial]” commit. This commit is important, because it gives purists an avenue to revert modernizing changes to the original text."

That's pretty much a hard-and-fast rule from Alex. Rebasing the "modernize spellings" commit to be prefixed with [Editorial] won't be a big deal. But for your proofreading commits, they will need to be separated into Editorial commits for the spelling changes, and then regular commits for the other changes.

**Comment by kingrpaul** [comment]
Differences from step-by-step came about, in part, because sections were added one at a time from varying sources. Had some challenges keeping things from getting out of hand. I believe the first few commits would follow the labeling from step-by-step.
I am not sure I would be able to segregate those at this point.

**Comment by vr8hub** [comment]
Right, understood, I had 13 books to deal with on O. Henry's short stories, so I feel your pain. Whether X went before Y or whether Z was done four times isn't the issue here, though.

In the case of the editorial commits, it's 1) whether they're labelled correctly, 2) whether they contain only editorial changes, and 3) whether all editorial-type changes are done in an Editorial commit.

And I'm just the messenger. :) As I said, Alex has historically held that, as stated in the Step-by-Step, all three of the above items have to be true. You are welcome (as with everything) to ask him about it on the list; as Editor-in-Chief he's the final arbiter.

**Comment by kingrpaul** [comment]
Fair enough. This may have been over-ambitious as project #2. I'll give it some thought.

**Comment by vr8hub** [comment]
Possibly. But you've done the majority of the work already, so it would be a shame not to finish it. Just don't feel like you have to finish it this week. Or even this month. :) Take as much time as you need. This is a volunteer project, so there's no deadline.

In this specific case, it shouldn't be _too_ difficult to fix this. Warning: I am not a git expert by any means, I've learned it because of this project, so continue at your own risk; there are several experts on the list, so ask there if you need more help, or you can always ask on Stackexchange.
* Make a couple of copies of your repository. Now you have three: the original and copy1/copy2; don't touch the original, it's your safety net.
* In copy1, do a rebase to rename the "modernize spellings" commit to "[Editorial] Modernize spelling". (I'm assuming that commit is where you ran `se modernize-spelling`?)
* In copy1, then revert back to before your first proofreading commit. You can find instructions for rolling back commits in a web search, but I've found [git-extras](https://github.com/tj/git-extras) invaluable for this and other reasons. It has a `git undo X` command that will roll back as many commits as you like; it doesn't do anything different, it's just syntactical sugar that is a lot easier to remember than the actual git command. You want a hard undo; you want the files to be as of that commit, without any of the changes from future commits.
* On copy1, run the various spelling fixes from the step-by-step guide, commiting each as instructed there.
* On copy2, checkout your second proofreading commit. (Just checkout, don't rollback.)
* On copy2, run the same spelling commands from the step-by-step that you ran on copy1. You don't need to commit them, just run them. This is in case those commands find more fixes than you've made manually; we don't want to lose them in what we're doing next.
* Copy the story files from copy2 over to copy1.

The differences in copy1 are now going to be the non-spelling changes, since you already made the spelling changes in copy1. It was three commits initially (proofreading, de-lint, and proofreading), but I don't see any problem with them being a single "Proofreading" commit.

For the rest of the commits after the second proofreading one, you can do a diff on copy2 to see the differences from each commit, and make those changes in copy1, just making sure you're not losing any spelling fixes that happen to be in something else you've changed. (By then the chances should be slim, but best to be sure.) It's possible to create a `git patch` script in copy2 and then apply it in copy1; you can do a web search if you want to go crazy. :)

If all goes well, when you're finished copy1 will contain the editorial commits as needed, along with your other changes. After making a backup or three of your original, you can copy copy1 over the original, and push them up to GitHub. (At that point, it's probably going to be better/easier to delete the GitHub repository and create a new identically-named one.)
If it doesn't go well, you have the original and can start over with new copies. :)

**Comment by kingrpaul** [comment]
Thank you. My git skills don't go very deep. I will need to look more closely into re-basing and git-extras. There was a modernize-spelling late in the process though I believe some sections were modernized along the way, which was the intent. I'll read this again with fresh eyes tomorrow. Much appreciated.

**Comment by vr8hub** [comment]
OK. The general rule is that anywhere the spelling is modernized has to be in an Editorial commit, so I'll leave you to figuring out what applies where.
